WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -87,12 +89,47 @@ protocol ProductImageUploaderProtocol {
 
 /// Supports background image upload and product images update after the user leaves the product form.
 final class ProductImageUploader: ProductImageUploaderProtocol {
+
+    let imageStatusStorage: ProductImageStatusStorage
+
     var errors: AnyPublisher<ProductImageUploadErrorInfo, Never> {
-        errorsSubject.eraseToAnyPublisher()
+        if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+            return imageStatusStorage.errorsPublisher
+                .flatMap { errorItems in
+                    errorItems.publisher
+                        .compactMap { errorItem in
+                            guard let productOrVariationID = errorItem.productOrVariationID,
+                                  let assetType = errorItem.assetType else { return nil }
+                            return ProductImageUploadErrorInfo(
+                                siteID: errorItem.siteID,
+                                productOrVariationID: productOrVariationID,
+                                error: .failedUploadingImage(asset: assetType, error: errorItem.error)
+                            )
+                        }
```

Comment:
I also missed in the previous reviews that the stream should exclude any keys listed in `statusUpdatesExcludedProductKeys`. That way, we would not display the error notice when an upload fails while the user is on the same product form.

<img src="https://github.com/user-attachments/assets/b57a3aff-7fa9-468e-8aca-4b807dbf8fb3" width=320 />

Notice above that both the error notice and alert are displayed.

Reply:
Nice catch Huong, I fixed it here 5b719e4e693208eb07bf80d71efa83b2ca51a532 can you please take another look?